doc/linecard_hotswap/LineCard_hotSwap_HLD.md

```diff
@@ -0,0 +1,419 @@
+* [3 Design](#design)  
+    * [3.1 Architecture](#architecture)  
+    * [3.2 OrchAgent Changes](#orchagent-changes) 
+    * [3.3 Syncd Changes](#syncd-changes) 
+    * [3.4 DB Changes](#db-changes)  
+        * [3.4.1 State DB](#state-db)
+    * [3.5 SAI](#sai)  
+    * [3.6 CLI](#cli)
+* [4 Flows](#flows)  
+    * [4.1 Initialization](#initialization)  
+        * [4.1.1 Event Callback Registration](#event-callback)
+        * [4.1.2 Port State Table](#port-state-table)
+        * [4.1.3 Port initialize](#port-initialize)
+    * [4.2 Line Card Hot Event Notification](#notification)  
+    * [4.3 Re-apply Configuration on Line Card](#reapply)  
+* [5 Serviceability and Debug](#serviceability)
+* [6 Warm Boot Support](#warm-boot)
+* [7 Scalability](#scalability)
+* [8 Tests](#tests)  
+    * [8.1 Unit tests](#unit-test)  
+* [9 Unsupported features](#unsupported)
+
+## About this manual<a name="about-this-manual"></a>
+This document describes the software high-level design of line card hot swapping implementation in SONiC.
+
+## Revision<a name="revision"></a>
+
+| Rev | Date       | Author         | Description                       |
+|:---:|:----------:|:--------------:|:----------------------------------|
+| 0.1 | 05/02/2020 | Fish, Steve    | Initial version                   |
+
+## Abbreviation<a name="abbreviation"></a>
+
+|Term	|Meaning                     |
+|:-----:|:--------------------------:|
+|PHY	|Physical layer              |
+|DB     |Date base                   |
+|CLI    |Command-Line Interface      |
+|IPC	|Inter Process communication |
+
+## List of figures<a name="list-of-figures"></a>
+[Figure 1: Line Card Hot Swap High Level Architecture](./f1.jpg)  
+[Figure 2: Event callback registration](./f2.jpg)  
+[Figure 3: State table update](./f3.jpg)  
+[Figure 4: Port Initialize_1](./f4.jpg)  
+[Figure 5: Port Initialize_2](./f5.jpg)  
+[Figure 6: Line Card Event Notification](./f6.jpg)  
+[Figure 7: Line Card Event Notification Flow](./f7.jpg)  
+[Figure 8: Reapply Configuration flow](./f8.jpg)  
+
+## 1 Introduction and Scope<a name="introduction"></a>
+
+A modern network switch in data center must be a high fault tolerance equipment which consists of many swappable hardware components (e.g., PSU/FAN/QSFP). 
+For stability, it is not allowed to reboot the device after hot-swapping hardware components. The swappable design is very important for data center to minimize the impact of failures on certain hardware components on the switch. 
+This HLD provides a generic software mechanism of line card hot swapping, which includes the architecture, functional flow and CLI command.
+
+## 2 Requirements<a name="requirements"></a>
+
+### 2.1 Functional Requirements<a name="functionality"></a>
+
+**This feature will support the following functionality:**
+1.	1 chassis consists of swappable line cards.
+2.	Every line card with the same card type, 32*100G(Different card type will be implemented in the near future.)
```

Comment:
please have a clear definition of line card.

Reply:
what is the requirement of phy driver, does it requires firmware download for the phy driver.

Reply:
OK, we will define what is included in the line card.

Reply:
We update the definition of line card. For the requirement of phy driver, we can based on  gearbox design to use a profile while PHY initializes. I will integrate our design with #538